src/main/java/org/symphonyoss/symphony/entityjson/EntityJsonParser.java

```diff
@@ -0,0 +1,283 @@
+/*
+ *
+ *
+ * Copyright 2017 Symphony Communication Services, LLC.
+ *
+ * Licensed to The Symphony Software Foundation (SSF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The SSF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.symphonyoss.symphony.entityjson;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.fge.jsonschema.core.exceptions.ProcessingException;
+import com.github.fge.jsonschema.core.load.configuration.LoadingConfiguration;
+import com.github.fge.jsonschema.core.load.configuration.LoadingConfigurationBuilder;
+import com.github.fge.jsonschema.main.JsonSchemaFactory;
+import com.github.fge.jsonschema.main.JsonSchemaFactoryBuilder;
+
+/**
+ * A parser for EntityJSON and StructuredObject.
+ * 
+ * Obtain via EntityJsonParserBuilder.
+ * 
+ * @author Bruce Skingle
+ *
+ */
+public class EntityJsonParser
+{
+  public static final URL JSON_SCHEMA_SCHEMA_URL        = getURL("/schema/org/json-schema/draft-04/schema.json");
+  public static final URL ENTITY_JSON_SCHEMA_URL        = getURL("/schema/entity-json-v0_1.json");
+  public static final URL ENTITY_JSON_EXAMPLE_URL       = getURL("/example/entity-json-01.json");
+  public static final URL STRUCTURED_OBJECT_SCHEMA_URL  = getURL("/schema/structured-object-v0_1.json");
+  public static final URL BOND_RFQ_SCHEMA_URL           = getURL("/proposed/org/symphonyoss/fin/rfq/request/bond-v0_1.json");
+  public static final URL BOND_RFQ_EXAMPLE_URL          = getURL("/example/org/symphonyoss/fin/rfq/request/bond-01.json");
+  
+  private static URL getURL(String url)
+  {
+    try
+    {
+      return new URL("https://symphonyosf.github.io/symphony-object" + url);
+    }
+    catch (MalformedURLException e)
+    {
+      throw new RuntimeException("Invalid URL", e);
+    }
+  }
+
+  private final JsonSchemaFactory factory_;
+  
+  /* package */ EntityJsonParser(boolean unrestrictedSchemaLoad)
+  {
+    if(unrestrictedSchemaLoad)
```

Comment:
I'm not commenting on code style in general as it's (largely) personal preference, but I am strongly opposed to braceless _if-else_s. I like this explanation for why this is bad, bug-prone practice: https://www.joelonsoftware.com/2005/05/11/making-wrong-code-look-wrong/

Reply:
That blog is frankly a load of tosh, but since I refuse to argue about curly braces I added two for you.

I even put one them in the wrong place to avoid a followup argument.